travis: skip install step

alextoombs opened the pull request:
Listing `install: true` skips the install step, both the explicit and
implicit commands called by Travis during the build.

Remove Go 1.4 from the set of runtimes we test against.

Also, add the `-race` flag.

@levb @alekstkach

levb commented:
LGTM

philpennock commented:
Why are you getting the repo `github.com/apcera/gssapi` manually?  That will just create confusion about whether you're testing a given branch or existing master status.

alextoombs commented:
Good question; it seems like if you remove the `install` step from the `.travis.yml`, Travis will implicitly run `go get -t -v ./...`. See [this build](https://travis-ci.org/apcera/gssapi/jobs/122005807); unfortunately, I squashed and force-pushed, so you can't see the diff. However, it was failing consistently when I removed the `install` step.

I'm not entirely sure why, or how I could just disable the step. You can see further above in the step that we're explicitly cloning the repo and checking out the PR merge ref.

philpennock commented:
So the difference is `-t` which says to get the stuff required for tests, so goes looking at stuff in the `test` namespace.

In the `test` dir, `Context` comes from `main_test.go`; for some reason I'm not immediately spotting, that's not getting picked up.  AFAIK it _should_ still be part of the one package, since it's called `test`, so all the files are built together, but that's not happening.

Renaming `test/main_test.go` to `test/main.go` fixes the problem.

IMO it'd be much better to do the file rename and keep to just one copy of the package on local disk, to avoid any ambiguity about which is being tested.

alextoombs commented:
Good suggestion; I'll make that change, thanks.

alextoombs commented:
The file rename has a side effect wherein the `Context` is not initialized in the `test` package for all tests, so it's panicking reliably.

philpennock commented:
Huh, didn't panic for me with the tests I invoked but I didn't bring up Docker stuff.

Will drink coffee and think on it.

alextoombs commented:
I ended up not renaming the file due to the global state issue; it's the right change, but it's a non-trivial amount of work. Instead, I'm doing the following:

1. Explicitly [skipping the install step](https://docs.travis-ci.com/user/customizing-the-build/)

2. Removing Go 1.4
  - FWIW, this has a [known issue where Go 1.4.3 builds are missing `go vet`](https://github.com/golang/go/issues/12883)

Will merge on green.
